### ISSUE TITLE
☄️ v1.10.0 Changelog

### ISSUE THREAD

**Issue by Fenny** [gofiber/fiber#416]
Dear 🐻 Gophers,

Fiber `v1.10.0` is finally released and it has some **important** changes we would like to share with you.
<br>

⚠️ **Breaking Changes**
-
- All routes will return the registered [`Route`](https://pkg.go.dev/github.com/gofiber/fiber?tab=doc#Route) metadata instead of `App`, chaining won't be possible anymore.
- ~~`ctx.Settings.TemplateEngine`~~
- ~~`ctx.Settings.TemplateFolder`~~
- ~~`ctx.Settings.TemplateExtension`~~
Are replaced by the new [`ctx.Settings.Templates`](https://pkg.go.dev/github.com/gofiber/fiber?tab=doc#Templates) interface. See our updated [template](https://github.com/gofiber/template) middleware for examples with support for `8` template engines.
<details>
  <summary>📚 Show syntax</summary>

```go
type Engine struct {
	templates *template.Template
}

func (e *Engine) Render(w io.Writer, name string, data interface{}) error {
	return e.templates.ExecuteTemplate(w, name, data)
}

func main() {
	app := fiber.New()
	engine := &Engine{
		// ./views/index.html
		// <h1>{{.Title}}</h1>
		templates: template.Must(template.ParseGlob("./views/*.html")),
	}
	app.Settings.Template = engine

	app.Get("/hello", func(c *fiber.Ctx) {
		c.Render("index", fiber.Map{
			"Title": "Hello, World!",
		})
	})
}
```
</details>

- ~~[ctx.Body(key ...string)](https://docs.gofiber.io/context#body)~~
Accessing form data via `Body` is deprecated, please use `ctx.FormValue(key string)` instead.
- ~~[ctx.Cookies(key ...string)](https://docs.gofiber.io/context#cookies)~~
To get the raw `Cookie` header, please use `ctx.Get("Cookies")` instead.

🔥 **New**
-
- [`Handler`](https://pkg.go.dev/github.com/gofiber/fiber?tab=doc#Handler) type has been added
- [app.Add](https://docs.gofiber.io/application#http-methods) register handlers with a HTTP method input: `app.Add("GET", "/foo", handler)`.
- [app.Settings.Templates]((https://pkg.go.dev/github.com/gofiber/fiber?tab=doc#Templates) is the interface that wraps the Render function.
- [app.Settings.DisableHeaderNormalizing](https://docs.gofiber.io/application#settings) Disable normalization: `conteNT-tYPE -> Content-Type`
- [`ctx.Context()`](https://pkg.go.dev/github.com/gofiber/fiber?tab=doc#Ctx.Context) returns context.Context that carries a deadline, a cancellation signal, and other values across API boundaries. https://github.com/gofiber/fiber/pull/383
<br>

🩹 **Fixes**
-
- Treat exact param keys in request path as params https://github.com/gofiber/fiber/issues/405
- `c.Params()` now accepting case sensitive values and keys https://github.com/gofiber/fiber/issues/392
- Cookies `SameSite` attribute defaults to `Lax` if not set
- `c.Append()` does not append duplicates anymore on case-sensitive values
- `c.SendFile("./404.html")` would overwrite previous status codes, this has been fix. https://github.com/gofiber/fiber/issues/391
- [app.Test](https://docs.gofiber.io/application#test) Would throw an `EOF` error if you do not provide a `Content-Length` header when passing a body `io.Reader` with [NewRequest](https://golang.org/pkg/net/http/httptest/#NewRequest). This is not necessary anymore, it will add the header for you if not provided.
- `ctx.Protocol()` also checks the `"X-Forwarded-Protocol"`, `"X-Forwarded-Ssl"` and `"X-Url-Scheme"` headers
<br>

🧹 **Updates**
-
- [app.Use](https://docs.gofiber.io/application#http-methods) & [app.Group](https://docs.gofiber.io/application#group) now supports `/:params` & `/:optionals?` inside the prefix path.
- Fiber docs are now fully translated in [Russian](https://docs.gofiber.io/v/ru/) & [Chinese](https://docs.gofiber.io/v/cn/)
- Add new supporters to README's
- Update template examples in README's
- Add `./public` to Static examples in README's https://github.com/gofiber/fiber/issues/411
- Add new media articles to README's [Improve performance](https://dev.to/koddr/fiber-v1-9-5-how-to-improve-performance-by-817-and-stay-fast-flexible-and-friendly-2dp6) & [web-based authentication](https://vugt.me/the-road-to-web-based-authentication-with-fiber/)
- With the help of @renewerner87 we produce zero garbage on matching and dispatching incoming requests. The only heap allocations that are made, is by building the key-value pairs for path parameters. If the requested route contains no parameters, not a single allocation is necessary.
<br>

🧬 **Official Middlewares**
-
- [gofiber/utils](https://github.com/gofiber/utils) `v0.0.3`
- [gofiber/logger](https://github.com/gofiber/logger) `v0.0.9`
- [gofiber/session](https://github.com/gofiber/session) `v1.1.0`
- [gofiber/template](https://github.com/gofiber/template) `v1.3.0`
- [gofiber/websocket](https://github.com/gofiber/websocket) `v0.2.1`
<br>

🌱 **Third Party Middlewares**
-
- [thomasvvugt/fiber-boilerplate](https://github.com/thomasvvugt/fiber-boilerplate)
- [arsmn/gqlgen](https://github.com/arsmn/gqlgen)

**Comment by Fenny** [comment]
@invipal hyphen support is now available `v1.12.+`
```go
// http://localhost:3000/plantae/prunus.persica
app.Get("/plantae/:genus.:species", func(c *fiber.Ctx) {
  c.Params("genus")   // prunus
  c.Params("species") // persica
})

// http://localhost:3000/flights/LAX-SFO
app.Get("/flights/:from-:to", func(c *fiber.Ctx) {
  c.Params("from")   // LAX
  c.Params("to")     // SFO
})
```

**Comment by invipal** [comment]
Thank you @Fenny, with this feature, Fiber now fully met our expectation for developing our next project.
